.travis.yml

```diff
@@ -1,27 +1,91 @@
 language: go
 
-matrix:
-  include:
-  -  os: linux
-     sudo: required
-     services:
-       - docker
-  -  os: osx
-
 go:
   - 1.15.x
 
 notifications:
   email: false
 
-install:
-  # Ideally, the (brew update) should not be necessary and Travis would have fairly
-  # frequenstly updated OS images; that’s not been the case historically.
-  # In particular, explicitly unlink python@2, which has been removed from Homebrew
-  # since the last OS image build (as of July 2020), but the Travis OS still
-  # contains it, and it prevents updating of Python 3.
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew update && brew unlink python@2 && brew install gpgme ; fi
-
-script:
-  - if [[ "$TRAVIS_OS_NAME" == "osx" ]];   then hack/travis_osx.sh ; fi
-  - if [[ "$TRAVIS_OS_NAME" == "linux" ]]; then make vendor && ./hack/tree_status.sh && make local-cross && make check ; fi
+env:
+  global:
+    - REGISTRY=quay.io
+
+# Just declaration of the image-build-push step with script actions to execute
+x_base_steps:
+  - &image-build-push
+    services:
+      - docker
+    os: linux
+    script:
+      # skopeo upstream image build and push
+      - export SOURCE_TYPE=upstream; export REPO=skopeo/upstream; export TAG=master
+      - make build-arch-container
+      # Dont's push if it's PR check
+      - if [[ "$TRAVIS_EVENT_TYPE" != "pull_request" ]]; then make push-arch-container ; fi
```

Comment:
Could the two conditions here be positive (“is this a cron job / …”) instead of just excluding one type? This might do unexpected pushes if any new event types are added or set up.

Reply:
At this moment Travis has 4 cases here `push`, `pull_request`, `api`, `cron`. I can add the check for cron and push type

Reply:
added check for push and cron